--- farsight2-0.0.31/gst-libs/gst/farsight/fs-conference-iface.c.orig	2011-10-09 23:06:33.000000000 +0200
+++ farsight2-0.0.31/gst-libs/gst/farsight/fs-conference-iface.c	2021-03-08 18:39:40.693431414 +0100
@@ -43,7 +43,7 @@
  *
  * This will communicate asynchronous events to the user through #GstMessage
  * of type #GST_MESSAGE_ELEMENT sent over the #GstBus.
- * </para>
+ *
  * <refsect2><title>The "<literal>farsight-error</literal>" message</title>
  * |[
  * "src-object"       #GObject           The object (#FsConference, #FsSession or #FsStream) that emitted the error
@@ -55,7 +55,7 @@
  * The message is sent on asynchronous errors.
  * </para>
  * </refsect2>
- * <para>
+ *
  */
 
 static void fs_conference_iface_init (FsConferenceClass *iface);
--- farsight2-0.0.31/gst-libs/gst/farsight/fs-session.c.orig	2011-10-09 23:06:33.000000000 +0200
+++ farsight2-0.0.31/gst-libs/gst/farsight/fs-session.c	2021-03-08 18:40:29.326501279 +0100
@@ -39,7 +39,7 @@
  *
  * This will communicate asynchronous events to the user through #GstMessage
  * of type #GST_MESSAGE_ELEMENT sent over the #GstBus.
- * </para>
+ *
  * <refsect2><title>The "<literal>farsight-send-codec-changed</literal>"
  *   message</title>
  * |[
@@ -68,7 +68,7 @@
  * next "farsight-codecs-changed" message until reading the codecs.
  * </para>
  * </refsect2>
- * <para>
+ *
  */
 
 #ifdef HAVE_CONFIG_H
--- farsight2-0.0.31/gst-libs/gst/farsight/fs-stream.c.orig	2011-10-09 23:06:33.000000000 +0200
+++ farsight2-0.0.31/gst-libs/gst/farsight/fs-stream.c	2021-03-08 18:41:26.726190318 +0100
@@ -36,7 +36,7 @@
  *
  * This will communicate asynchronous events to the user through #GstMessage
  * of type #GST_MESSAGE_ELEMENT sent over the #GstBus.
- * </para>
+ *
  * <refsect2><title>The "<literal>farsight-new-local-candidate</literal>" message</title>
  * |[
  * "stream"           #FsStream          The stream that emits the message
@@ -93,7 +93,7 @@
  * This message is emitted the state of a component of a stream changes.
  * </para>
  * </refsect2>
- * <para>
+ *
  */
 
 #ifdef HAVE_CONFIG_H
--- farsight2-0.0.31/gst/fsrtpconference/fs-rtp-session.c.orig	2021-03-08 22:31:45.561327262 +0100
+++ farsight2-0.0.31/gst/fsrtpconference/fs-rtp-session.c	2021-03-08 22:31:55.317941073 +0100
@@ -30,7 +30,7 @@
  * fs_conference_new_session() on a #FsRtpConference. It can be either
  * Audio or Video. It also represents data send with one and only one
  * SSRC (although if there is a SSRC collision, that SSRC may change).
- * </para>
+ *
  * <refsect2><title>Codec profiles</title>
  * <para>
  * It is possible to define "codec profiles", that is non-autodetected
@@ -50,7 +50,7 @@
  * Also, it is possible to declare profiles with only a decoding pipeline,
  * you will only be able to receive from this codec, the encoding may be a
  * secondary pad of some other codec.
- * </para></refsect2><para>
+ * </para></refsect2>
  */
 
 #ifdef HAVE_CONFIG_H
--- farsight2-0.0.31/common/gstdoc-scangobj.orig	2010-10-08 18:38:56.000000000 +0200
+++ farsight2-0.0.31/common/gstdoc-scangobj	2021-03-08 21:29:32.704883215 +0100
@@ -30,9 +30,6 @@
 
 use Getopt::Long;
 
-unshift @INC, '/usr/share/gtk-doc/data';
-require "gtkdoc-common.pl";
-
 # Options
 
 # name of documentation module
@@ -1791,6 +1788,59 @@
   unlink "./$MODULE-scan.c", "./$MODULE-scan.o", "./$MODULE-scan.lo", "./$MODULE-scan";
 }
 
+# Copied from gtk-doc 1db161bd708cdfb88b362ea0b5d047034d9c3272
+#############################################################################
+# Function    : UpdateFileIfChanged
+# Description : Compares the old version of the file with the new version and
+#                if the file has changed it moves the new version into the old
+#                versions place. This is used so we only change files if
+#                needed, so we can do proper dependency tracking and we don't
+#                needlessly check files into version control systems that haven't
+#               changed.
+#                It returns 0 if the file hasn't changed, and 1 if it has.
+# Arguments   : $old_file - the pathname of the old file.
+#                $new_file - the pathname of the new version of the file.
+#                $make_backup - 1 if a backup of the old file should be kept.
+#                        It will have the .bak suffix added to the file name.
+#############################################################################
+
+sub UpdateFileIfChanged {
+    my ($old_file, $new_file, $make_backup) = @_;
+
+    #@TRACE@("Comparing $old_file with $new_file...");
+
+    # If the old file doesn't exist we want this to default to 1.
+    my $exit_code = 1;
+
+    if (-e $old_file) {
+        `cmp -s "$old_file" "$new_file"`;
+        $exit_code = $? >> 8;
+        #@TRACE@("   cmp exit code: $exit_code ($?)");
+    }
+
+    if ($exit_code > 1) {
+        die "Error running 'cmp $old_file $new_file'";
+    }
+
+    if ($exit_code == 1) {
+        #@TRACE@("   files changed - replacing old version with new version.");
+        if ($make_backup && -e $old_file) {
+            rename ($old_file, "$old_file.bak")
+                || die "Can't move $old_file to $old_file.bak: $!";
+        }
+        rename ($new_file, $old_file)
+            || die "Can't move $new_file to $old_file: $!";
+
+        return 1;
+    } else {
+        #@TRACE@("   files the same - deleting new version.");
+        unlink ("$new_file")
+            || die "Can't delete file: $new_file: $!";
+
+        return 0;
+    }
+}
+
 #&UpdateFileIfChanged ($old_signals_filename, $new_signals_filename, 0);
 &UpdateFileIfChanged ($old_hierarchy_filename, $new_hierarchy_filename, 0);
 &UpdateFileIfChanged ($old_interfaces_filename, $new_interfaces_filename, 0);
--- farsight2-0.0.31/docs/plugins/gtkdoc-scangobj-transmitters.orig	2010-11-01 22:32:09.000000000 +0100
+++ farsight2-0.0.31/docs/plugins/gtkdoc-scangobj-transmitters	2021-03-08 21:33:25.266956651 +0100
@@ -30,9 +30,6 @@
 
 use Getopt::Long;
 
-unshift @INC, '/usr/share/gtk-doc/data';
-require "gtkdoc-common.pl";
-
 # Options
 
 # name of documentation module
@@ -842,7 +839,7 @@
 
   for (i = 0; i < level; i++)
     fprintf (fp, "  ");
-  fprintf (fp, g_type_name (type));
+  fprintf (fp, "%s", g_type_name (type));
   fprintf (fp, "\\n");
 
   children = g_type_children (type, &n_children);
@@ -893,7 +890,7 @@
 
   if (n_interfaces > 0)
     {
-      fprintf (fp, g_type_name (type));
+      fprintf (fp, "%s", g_type_name (type));
       for (i=0; i < n_interfaces; i++)
           fprintf (fp, " %s", g_type_name (interfaces[i]));
       fprintf (fp, "\\n");
@@ -938,7 +935,7 @@
 
   if (n_prerequisites > 0)
     {
-      fprintf (fp, g_type_name (type));
+      fprintf (fp, "%s", g_type_name (type));
       for (i=0; i < n_prerequisites; i++)
           fprintf (fp, " %s", g_type_name (prerequisites[i]));
       fprintf (fp, "\\n");
@@ -1558,6 +1555,59 @@
 
 unlink "./$MODULE-scan.c", "./$MODULE-scan.o", "./$MODULE-scan.lo", "./$MODULE-scan";
 
+# Copied from gtk-doc 1db161bd708cdfb88b362ea0b5d047034d9c3272
+#############################################################################
+# Function    : UpdateFileIfChanged
+# Description : Compares the old version of the file with the new version and
+#                if the file has changed it moves the new version into the old
+#                versions place. This is used so we only change files if
+#                needed, so we can do proper dependency tracking and we don't
+#                needlessly check files into version control systems that haven't
+#               changed.
+#                It returns 0 if the file hasn't changed, and 1 if it has.
+# Arguments   : $old_file - the pathname of the old file.
+#                $new_file - the pathname of the new version of the file.
+#                $make_backup - 1 if a backup of the old file should be kept.
+#                        It will have the .bak suffix added to the file name.
+#############################################################################
+
+sub UpdateFileIfChanged {
+    my ($old_file, $new_file, $make_backup) = @_;
+
+    #@TRACE@("Comparing $old_file with $new_file...");
+
+    # If the old file doesn't exist we want this to default to 1.
+    my $exit_code = 1;
+
+    if (-e $old_file) {
+        `cmp -s "$old_file" "$new_file"`;
+        $exit_code = $? >> 8;
+        #@TRACE@("   cmp exit code: $exit_code ($?)");
+    }
+
+    if ($exit_code > 1) {
+        die "Error running 'cmp $old_file $new_file'";
+    }
+
+    if ($exit_code == 1) {
+        #@TRACE@("   files changed - replacing old version with new version.");
+        if ($make_backup && -e $old_file) {
+            rename ($old_file, "$old_file.bak")
+                || die "Can't move $old_file to $old_file.bak: $!";
+        }
+        rename ($new_file, $old_file)
+            || die "Can't move $new_file to $old_file: $!";
+
+        return 1;
+    } else {
+        #@TRACE@("   files the same - deleting new version.");
+        unlink ("$new_file")
+            || die "Can't delete file: $new_file: $!";
+
+        return 0;
+    }
+}
+
 #&UpdateFileIfChanged ($old_signals_filename, $new_signals_filename, 0);
 &UpdateFileIfChanged ($old_hierarchy_filename, $new_hierarchy_filename, 0);
 &UpdateFileIfChanged ($old_interfaces_filename, $new_interfaces_filename, 0);
--- farsight2-0.0.31/common-modified/gtk-doc-plugins.mak.orig	2011-09-07 22:20:18.000000000 +0200
+++ farsight2-0.0.31/common-modified/gtk-doc-plugins.mak	2021-03-08 22:25:41.416633337 +0100
@@ -156,23 +156,13 @@ scan-build.stamp: $(HFILE_GLOB) $(EXTRA_
 	    --ignore-headers="$(IGNORE_HFILES)";			\
 	touch scan-build.stamp
 
-#### update templates; done on every build ####
+#### generate templates; done on every build ####
 
-### FIXME: make this error out again when docs are fixed for 0.9
-# in a non-srcdir build, we need to copy files from the previous step
-# and the files from previous runs of this step
-tmpl-build.stamp: $(DOC_MODULE)-decl.txt $(SCANOBJ_FILES) $(DOC_MODULE)-sections.txt $(DOC_OVERRIDES)
-	@echo '*** Rebuilding template files ***'
-	if test x"$(srcdir)" != x. ; then				\
-	    for f in $(SCANOBJ_FILES) $(SCAN_FILES);			\
-	    do								\
-	        if test -e $(srcdir)/$$f; then cp $(srcdir)/$$f . ; fi; \
-	    done;							\
-	fi
-	gtkdoc-mktmpl --module=$(DOC_MODULE) | tee tmpl-build.log
+tmpl-build.stamp:
+	@echo '  DOC   Building template files'
 	$(PYTHON) \
 		$(top_srcdir)/common/mangle-tmpl.py $(srcdir)/inspect tmpl
-	@cat $(DOC_MODULE)-unused.txt
+	@-cat $(DOC_MODULE)-unused.txt
 	rm -f tmpl-build.log
 	touch tmpl-build.stamp
 
@@ -190,9 +180,13 @@ sgml-build.stamp: tmpl.stamp scan-build.
 		$(top_srcdir)/common/plugins.xsl $$a > xml/`basename $$a`; done
 	@for f in $(EXAMPLE_CFILES); do \
 		$(PYTHON) $(top_srcdir)/common/c-to-xml.py $$f > xml/element-`basename $$f .c`.xml; done
+	@_source_dir='' ;						\
+	for i in $(DOC_SOURCE_DIR) ; do					\
+	    _source_dir="$${_source_dir} --source-dir=$$i" ;	        \
+	done ;								\
 	gtkdoc-mkdb \
 		--module=$(DOC_MODULE) \
-		--source-dir=$(DOC_SOURCE_DIR) \
+		$${_source_dir} \
 		 --expand-content-files="$(expand_content_files)" \
 		--main-sgml-file=$(srcdir)/$(DOC_MAIN_SGML_FILE) \
 		--output-format=xml \
@@ -217,10 +211,7 @@ html-build.stamp: sgml.stamp $(DOC_MAIN_
 	@for f in $(content_files); do cp $(srcdir)/$$f html; done
 	cp -pr xml html
 	cp ../version.entities html
-	cd html && gtkdoc-mkhtml $(DOC_MODULE) $(DOC_MAIN_SGML_FILE)
-	mv html/index.sgml html/index.sgml.bak
-	$(SED) "s/ href=\"$(DOC_MODULE)\// href=\"$(DOC_MODULE)-@GST_MAJORMINOR@\//g" html/index.sgml.bak >html/index.sgml
-	rm -f html/index.sgml.bak
+	cd html && gtkdoc-mkhtml $(DOC_MODULE)-@GST_MAJORMINOR@ $(DOC_MAIN_SGML_FILE)
 	rm -f html/$(DOC_MAIN_SGML_FILE)
 	rm -rf html/xml
 	rm -f html/version.entities
@@ -269,11 +260,11 @@ install-data-local:
 	      $(INSTALL_DATA) $$i $(DESTDIR)$(TARGET_DIR); \
 	    done; \
 	  fi; \
-	  echo '-- Installing $(srcdir)/html/$(DOC_MODULE).devhelp' ; \
-	  $(INSTALL_DATA) $(srcdir)/html/$(DOC_MODULE).devhelp \
+	  echo '-- Installing $(srcdir)/html/$(DOC_MODULE)-@GST_MAJORMINOR@.devhelp' ; \
+	  $(INSTALL_DATA) $(srcdir)/html/$(DOC_MODULE)-@GST_MAJORMINOR@.devhelp \
 	    $(DESTDIR)$(TARGET_DIR)/$(DOC_MODULE)-@GST_MAJORMINOR@.devhelp; \
-	  if test -e $(srcdir)/html/$(DOC_MODULE).devhelp2; then \
-	            $(INSTALL_DATA) $(srcdir)/html/$(DOC_MODULE).devhelp2 \
+	  if test -e $(srcdir)/html/$(DOC_MODULE)-@GST_MAJORMINOR@.devhelp2; then \
+	            $(INSTALL_DATA) $(srcdir)/html/$(DOC_MODULE)-@GST_MAJORMINOR@.devhelp2 \
 	            $(DESTDIR)$(TARGET_DIR)/$(DOC_MODULE)-@GST_MAJORMINOR@.devhelp2; \
 	  fi; \
 	  (which gtkdoc-rebase >/dev/null && \
