--- farsight2-0.0.21/python/Makefile.am~	2009-05-08 23:34:14.000000000 +0200
+++ farsight2-0.0.21/python/Makefile.am	2011-01-23 02:41:42.365633977 +0100
@@ -1,5 +1,4 @@
 PYDEFS=`pkg-config --variable=defsdir pygobject-2.0`
-[ -z "$PYDEFS" && PYDEFS=`pkg-config --variable=defsdir pygtk-2.0`]
 GSTPYDEFS=`pkg-config --variable=defsdir gst-python-0.10`
 
 AM_CPPFLAGS =						\
--- farsight2-0.0.21/common/gtk-doc.mak~	2010-06-24 00:20:40.000000000 +0200
+++ farsight2-0.0.21/common/gtk-doc.mak	2011-01-23 02:46:59.972300645 +0100
@@ -232,6 +232,6 @@
 	-cp $(srcdir)/$(DOC_MODULE).types $(distdir)/
 	-cp $(srcdir)/$(DOC_MODULE)-sections.txt $(distdir)/
 	cd $(distdir) && rm -f $(DISTCLEANFILES)
-        -gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
+	-gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
 
 .PHONY : dist-hook-local docs
